src/rust/engine/process_execution/src/nailgun/nailgun_pool.rs

```diff
@@ -119,139 +122,175 @@ impl NailgunPool {
       jdk_path.clone()
     ));
 
-    Self::materialize_workdir_for_server(
-      store, workdir_path.clone(), jdk_path, input_files, workunit_store
-    ).and_then(move |_| {
-      debug!("Locking nailgun process pool so that only one can be connecting at a time.");
-      let mut processes = processes.lock();
-      debug!("Locked!");
-      let connection_result = if let Some(process) = processes.get_mut(&name) {
-        // Clone some fields that we need for later
-        let (process_name, process_fingerprint, process_port, build_id_that_started_the_server) = (
-          process.name.clone(),
-          process.fingerprint.clone(),
-          process.port,
-          process.build_id.clone(),
-        );
-
-        debug!(
-          "Checking if nailgun server {} is still alive at port {}...",
-          &process_name, process_port
-        );
+    let processes = self.processes.clone();
+    processes.lock()
+      .map_err(|_| "Failed to lock processes Mutex".to_string())
```

Comment:
Worth including the error in this string via `format!("Failed to lock processes Mutex: {:?}", e)`? Here and elsewhere.